pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -837,19 +1325,69 @@ func (pl *DynamicResources) PreBindPreFlight(ctx context.Context, cs fwk.CycleSt
 	if err != nil {
 		return statusError(klog.FromContext(ctx), err)
 	}
-	if len(state.claims) == 0 {
+	if state.claims.empty() {
 		return fwk.NewStatus(fwk.Skip)
 	}
 	return nil
 }
 
+// bindExtendedResources creates the requestMappings for the special extended resource claim
+func (pl *DynamicResources) bindRequestMappings(claim *resourceapi.ResourceClaim, pod *v1.Pod) []v1.ContainerExtendedResourceRequest {
```

Comment:
The doc comment doesn't actually tell me anything about what a `requestMapping` is. This method and the logic around `podScalarResources` remain a mystery to me and completely lack comments in the body which explain what the code does and why.

 `podScalarResources` is document as "May have extended resource backed by DRA." which is very vague. It's apparently a sum of all resources requested anywhere in the pod? At least say that.

This looks like an area where we need lots of dedicated unit tests which exercise the different ways how a pod may request resources. For example, I saw some comments in the called code about "pod level resources", but no corresponding unit tests. Should those be handled?

Reply:
added unittests for two functions: createRequestMappings, and createDeviceRequests. for "pod level resources", we don't need to care about them, as they can be only cpu or memory currently. (https://kubernetes.io/docs/tasks/configure-pod-container/assign-pod-level-resources/)